xc7/fasm2bels/verilog_modeling.py

```diff
@@ -31,6 +31,37 @@
 from .connection_db_utils import get_wire_pkey
 
 
+def pin_to_wire_and_idx(pin):
+    """ Break pin name into wire name and vector index.
+
+    Arguments
+    ---------
+    pin : str
+        Pin name, with optional vector index.
+
+    Returns
+    -------
+    wire : str
+        Wire name
+    idx : int
+        Vector index
+
+    >>> pin_to_wire_and_idx('A')
+    ('A', None)
+    >>> pin_to_wire_and_idx('A[0]')
+    ('A', 0)
+    >>> pin_to_wire_and_idx('A[1]')
+    ('A', 1)
+
+    """
+    idx = pin.find('[')
```

Comment:
I think we already have this function somewhere.
I also think a regex might have been better here.

The function is so small I don't really care however.